### PR TITLE
test(content): playwright refactor

### PR DESCRIPTION
**Pull request by divyajindel**
### Proposed behaviour

- Copy `content` folder from `/cypress/locators/` to `/playwright/components/`
- Rename `index.js` to `index.ts` and `locators.js` to `locators.ts` in `/playwright/components/content/` folder
- Refactor `index.ts` to Playwright
- Copy `content.cy.tsx from `/cypress/components/content/` to `/src/components/content/`
- Rename `content.cy.tsx` to `content.pw.tsx` in `/src/components/content/` folder
- Refactor tests to Playwright

### Current behaviour

`Content` tests are in Cypress not Playwright.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- [x] Run Content tests in Playwright Test Runner to check if the `content.pw.tsx` file passed
- [x] Run all other tests in Playwright Test Runner to check none of the other `*.pw.tsx` files have regressed

### CODE REVIEW

File: src/components/content/content.pw.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+const totalWidth = 1366;
+const alignmentData = ["left", "center", "right"];
+
+test.describe("check Content component properties", () => {
+  [
+    ["primary", "rgba(0, 0, 0, 0.9)"],
+    ["secondary", "rgba(0, 0, 0, 0.55)"],
+  ].forEach(([variant, titleColor]) => {
+    test(`should check ${variant} as variant for Content components ${titleColor}`, async ({
```
Comment (stephenogorman) [inline src/components/content/content.pw.tsx:23]: I don't think ${titleColor} is necessary in this test title.

File: src/components/content/content.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+  alignmentData.forEach((alignment) => {
+    test(`should check ${alignment} as alignment for Content component`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ContentComponent align={alignment} />);
+
+      const contentElement = contentTitle(page);
+      await expect(contentElement).toHaveCSS("text-align", `${alignment}`);
```
Comment (stephenogorman) [inline src/components/content/content.pw.tsx:75]: I don't think you need to enclose the alignment variable in quotes, the following should work:
`await expect(contentElement).toHaveCSS("text-align", alignment);`

File: src/components/content/content.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        const cssWidth = await getStyle(contentTitle(page), "width");
+        await expect(parseInt(cssWidth)).toBeLessThanOrEqual(
+          computedWidth - 30
+        );
+        await expect(parseInt(cssWidth)).toBeGreaterThanOrEqual(
+          computedWidth - 31
+        );
+      });
+    }
+  );
```
Comment (stephenogorman) [inline src/components/content/content.pw.tsx:120]: The refactor to Playwright has been done correctly for this test, but I think the original Cypress test is testing the wrong thing. It's supposed to be verifying that the `bodyFullWidth` prop sets the content body width to its maximum, but is actually testing the same thing as the previous test (content Title). The following actually tests the bodyFullWidth prop:

```
  ([
    [true, 70, totalWidth],
    [false, 50, totalWidth],
  ] as [boolean, number, number][]).forEach(
    ([bodyFullWidth, titleWidth]) => {
      test(`should check Content component has bodyFullWidth as ${bodyFullWidth}`, async ({
        mount,
        page,
      }) => {
        await mount(
          <ContentComponent
            bodyFullWidth={bodyFullWidth}
            titleWidth={titleWidth}
          />
        );
        const cssWidth = await getStyle(contentBody(page), "width");
        if (bodyFullWidth === true) {
          await expect(parseInt(cssWidth)).toEqual(totalWidth);
        } else {
          await expect(parseInt(cssWidth)).not.toEqual(totalWidth);
        }
      });
    }
  );
```

**Review by stephenogorman — CHANGES_REQUESTED** [review]

**Review by stephenogorman — DISMISSED** [review]

File: src/components/content/content-test.stories.tsx [hunk 4]
```diff
@@ -53,15 +53,3 @@ DefaultStory.args = {
   titleWidth: "",
   bodyFullWidth: false,
 };
-
-export const ContentComponentTest = ({
```
Comment (tomdavies73) [inline src/components/content/content-test.stories.tsx:57]: **suggestion**: You can add this back, there's not much reason for it to be deleted. Keeping the test story means people can still use it to test locally.

File: src/components/content/content.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  [
+    ["primary", "rgba(0, 0, 0, 0.9)"],
+    ["secondary", "rgba(0, 0, 0, 0.55)"],
+  ].forEach(([variant, titleColor]) => {
+    test(`should check ${variant} as variant for Content components`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ContentComponent variant={variant} />);
+      const contentElement = contentTitle(page);
```
Comment (tomdavies73) [inline src/components/content/content.pw.tsx:28]: **suggestion**: the `contentElement`, `const` doesn't need to be declared here. You can just include the `contentTitle` locator within the `expect` underneath, reducing a line of code like so `await expect(contentTitle(page)).toHaveCSS("color", `${titleColor}`);`

File: src/components/content/content.pw.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+  testData.forEach((children) => {
+    test(`should check ${children} as children for Content component`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <ContentComponent title="Title">{children}</ContentComponent>
+      );
+
+      await expect(await contentBody(page)).toHaveText(children);
```
Comment (tomdavies73) [inline src/components/content/content.pw.tsx:42]: **issue**: there is an unnecessary `await` within this expression

Comment (tomdavies73) [reply src/components/content/content.pw.tsx:42]: **suggestion**: just use one await before the `expect` like this `await expect(contentBody(page)).toHaveText(children);`

File: src/components/content/content.pw.tsx [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+  });
+
+  testData.forEach((title) => {
+    test(`should check ${title} as title for Content component`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ContentComponent title={title} />);
+
+      await expect(await contentTitle(page)).toHaveText(title);
```
Comment (tomdavies73) [inline src/components/content/content.pw.tsx:53]: **issue:** same comment as above, another unnecessary `await`

File: src/components/content/content.pw.tsx [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+    [70, (totalWidth * 70) / 100],
+    [50, (totalWidth * 50) / 100],
+  ] as [number, number][]).forEach(([titleWidth, computedWidth]) => {
+    test(`should check titleWidth as ${titleWidth} for Content component`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ContentComponent titleWidth={titleWidth} />);
+      const cssWidth = await getStyle(contentTitle(page), "width");
+      await expect(parseInt(cssWidth)).toBeLessThanOrEqual(computedWidth - 30);
```
Comment (tomdavies73) [inline src/components/content/content.pw.tsx:89]: **issue**: unnecessary `await` x2

Comment (tomdavies73) [reply src/components/content/content.pw.tsx:89]: **suggestion**: remove the unnecessary awaits, like so `expect(parseInt(cssWidth)).toBeLessThanOrEqual(computedWidth - 30) expect(parseInt(cssWidth)).toBeGreaterThanOrEqual(computedWidth - 31 );`

File: src/components/content/content.pw.tsx [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+    }) => {
+      await mount(
+        <ContentComponent
+          bodyFullWidth={bodyFullWidth}
+          titleWidth={titleWidth}
+        />
+      );
+      const cssWidth = await getStyle(contentBody(page), "width");
+      if (bodyFullWidth === true) {
+        await expect(parseInt(cssWidth)).toEqual(totalWidth);
```
Comment (tomdavies73) [inline src/components/content/content.pw.tsx:112]: **issue**: unnecessary `await` x 2

Comment (tomdavies73) [reply src/components/content/content.pw.tsx:112]: **suggestion**: remove the unnecessary awaits like so ` expect(parseInt(cssWidth)).toEqual(totalWidth);
      } else {
         expect(parseInt(cssWidth)).not.toEqual(totalWidth);
      }`

File: src/components/content/content.pw.tsx (hunk 6 above)
Comment (tomdavies73) [reply src/components/content/content.pw.tsx:42]: VS Code will let you know if an `await` is unnecessary by putting three dots underneath the `await` in question like so 
Screenshot 2023-08-18 at 12 48 20
Screenshot 2023-08-18 at 12 48 33

File: src/components/content/content.pw.tsx [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+    });
+  });
+
+  test(`should check inline parameter is enabled for Content component`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<ContentComponent inline />);
+
+    const contentElement = contentTitle(page);
```
Comment (tomdavies73) [inline src/components/content/content.pw.tsx:63]: **suggestion**: the `contentElement` `const` does not need to be defined here, you can just add the `contentTitle` locator within the `expect`. Similar comment to above

File: src/components/content/content.pw.tsx [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+  });
+
+  alignmentData.forEach((alignment) => {
+    test(`should check ${alignment} as alignment for Content component`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ContentComponent align={alignment} />);
+
+      const contentElement = contentTitle(page);
```
Comment (tomdavies73) [inline src/components/content/content.pw.tsx:74]: **suggestion**: the `contentElement` `const` does not need to be defined here, you can just add the `contentTitle` locator within the `expect`. Similar comment to above

**Review by tomdavies73 — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**